examples/calc.dfy

```diff
@@ -0,0 +1,62 @@
+module Old {
+  datatype expr = Const(x: int) | Add(e1: expr, e2: expr) | Sub(e1: expr, e2: expr)
+
+  function eval(e: expr): int {
+    match e {
+      case Const(n) => n
+      case Add(e1, e2) => eval(e1) + eval(e2)
+      case Sub(e1, e2) => eval(e1) - eval(e2)
+    }
+  }
+
+  lemma addcorrect(e1: expr, e2: expr)
+    ensures eval(Add(e1, e2)) == eval(Add(e2, e1))
+  {}
+}
+
+module New {
+
+  import Old
```

Comment:
Interesting, typeCart will identify the `expr` types of both modules as unequal. We could definitely update typeCart to generate the `tr` function.